src/structures/interfaces/TextBasedChannel.js

```diff
@@ -329,9 +334,14 @@ class TextBasedChannel {
    *   .then(interaction => console.log(`${interaction.customID} was clicked!`))
    *   .catch(console.error);
    */
-  awaitMessageComponentInteraction(options = {}) {
+  awaitMessageComponent(options = {}) {
+    const _options = {
+      ...options,
+      channel: this,
+      interactionType: InteractionTypes.MESSAGE_COMPONENT,
+    };
```

Comment:
🧹 This can go and pass `options` directly, as the two properties are assigned in it.

Reply:
It's actually missing `max: 1` but I wouldnt have noticed without this comment, so thanks.